File: src/main/java/com/example/finsentinel/service/trading/uta/ExecuteResult.java
package com.example.finsentinel.service.trading.uta;

import java.util.List;
import java.util.Map;

/**
 * Result of executing a trade commit — contains both the human-readable report
 * (for AI tool responses) and the structured data (for REST API responses).
 */
public record ExecuteResult(
        String report,
        Map<String, Object> commitData,
        List<Map<String, Object>> operationResults
) {}

File: apps/api/src/trading/interfaces/execute-result.ts
/**
 * Result of executing a pending commit.
 *
 * Result shape returned after executing a pending commit.
 * Contains a human-readable report, the raw commit data, and per-operation results.
 */
export interface ExecuteResult {
  /** Human-readable execution report. */
  report: string;

  /** The commit data that was executed (hash, message, timestamp, operations). */
  commitData: Record<string, unknown>;

  /** Per-operation execution results. */
  results: Record<string, unknown>[];
}
